.yarn/plugins/@yarnpkg/plugin-allow-scripts.cjs

```diff
@@ -0,0 +1,9 @@
+/* eslint-disable */
```

Comment:
Right, so I guess the question is, what benefit would shasums give us?

If it would aid in reviewing these pre-minified files, I'm not sure I agree. If one of these pre-minified files is different than a corresponding version in the module template for some reason, we'd want to know the differences. Right now, the best way to verify a file as a reviewer is to clone this repo, switch to this branch, clone the module template, do a manual diff between this file and the one in the module template, and verify that there is no output. I don't think that process would change if we had shasums; in fact, you'd have to manually diff the shasums file against the module template as well.

If, past the review process, it would serve as constant proof that the files aren't corrupted or have malicious code in them — after all, you wouldn't _need_ to clone the module template at this point, just pull it up alongside your repo on GitHub and compare the shasums file visually between the two repos — then I can see a small benefit, but then again, we'd have to set up some infrastructure to make it work well, and I'm not sure the juice would be worth the squeeze, as they say. We'd probably need to add a lint step to each repo that would verify that the shasum for a file is accurate. To do this, we'd want to regenerate the shasum for the file and ensure that it matches the shasum that has already been generated. However, what would this prove? It seems that what we really want to know is whether a pre-minified file matches the equivalent file that Yarn would download if you installed Yarn from scratch or installed a plugin from scratch. I guess we could figure out how to do this inside of this lint step, but that might be complicated, and I don't know that we ought to spend time figuring that out when we have other problems that we can solve.

I feel that the best place to address this would be in Yarn itself. In fact, if I'm reading the changelog correctly, Yarn 4 should help: 1) [pointing to a local Yarn executable will no longer be necessary if Corepack is installed](https://github.com/yarnpkg/berry/pull/4254) and 2) [missing plugins will be auto-installed by default](https://github.com/yarnpkg/berry/pull/4912) (and it will use checksums to determine whether a plugin is installed correctly or not). It looks like Yarn v4 has had an rc release for a year, so if we're really concerned about this, we could conceivably try it out across our repos. What do you think?

Reply:
In theory you could achieve the same thing using diffs by saying:

1. Run `yarn` against this branch. If either the binary or any of the plugin files are corrupted, Yarn will immediately throw an error. If that doesn't happen, continue.
2. Assuming that none of the files are corrupted, run `(cd ../metamask-module-template && find .yarn/plugins -type f -exec cp {} ../<project directory>/{} \;)`, then run `git diff`. If you see no changes, you can assume that all plugins match the module template correctly.
3. Run `yarn set version <whatever version is in .yarn/releases>`, e.g. `yarn set version 3.5.1`, then run `git diff`. If you see no changes, you can assume that this version is correct.

Something else I realized just now is that `.yarnrc.yml` has to be in sync with the plugin list in `.yarn/plugins` as well as the version set in `.yarn/releases`. While the approach above is a bit tedious, it has the advantage of:

- allowing projects to add whatever Yarn plugins they like (our `allow-scripts` plugin and Yarn's `constraints` plugin are required, but everything else is optional), and
- allowing projects to use whatever version of Yarn they like (in theory we could enforce a specific version, but that seems like an unnecessary burden short of automating it)

without having to keep `.yarnrc.yml` in sync as well.

Could we achieve the same thing with shasums?